packages/reactivity/src/collectionHandlers.ts

```diff
@@ -173,6 +173,24 @@ function createForEach(isReadonly: boolean, isShallow: boolean) {
   }
 }
 
+function createSetProtoMethod(method: TriggerOpTypes) {
+  return function (this: SetTypes, value: unknown, _isShallow = false) {
+    if (!_isShallow && !isShallow(value) && !isReadonly(value)) {
+      value = toRaw(value)
+    }
+    const target = toRaw(this)
+    const proto = getProto(target)
+    const hadKey = proto.has.call(target, value)
+    let result
+    if (!hadKey) {
+      // @ts-expect-error
+      result = target[method](value)
+      trigger(target, method, value, value)
```

Comment:
Shouldn't this be tracking rather than triggering?

These methods don't currently track correctly inside reactive effects. e.g.:

- [Playground - this PR](https://deploy-preview-11399--vue-sfc-playground.netlify.app/#eNp9Ustu2zAQ/JUFL5YBV0Yfp0A2mhY5tIe0aHrkRSI3NmOJFMilbUDQv2cpRYoDODlJnAc5s2Qnbts2P0YUN6IIypuWICDFdiutaVrnCTrwWCoyR1yBck0bCTX08OhdAwt2LqSVVjkbCErYzOLM4gkekLLlcqKr9+hJoFgwHZFlS9hsocyjNc5mVdIV6zEih+MFYdPWJSGvAIoqEjkL31Vt1GEjRZmXWmeLPda1Wyyl2N5qDZ+L9ai77qlGz8n5Ws+eL289ez9+tTluu44r932xTotLtLqKKkbhE9w7Aj7iYOzuQvOyb9pyrsz6VPqiqFgJCjysR7PLn4KzfG1dskmR5mZq9H9aYnOQ4gYGJnElz+D0e8DIR1xNuNqjOlzBn8I5YVL89RjQH1GKmaPS75BG+u7hHs/8P5ON07Fm9QfkPwyujinjKPsRrebYF7oh7a/h8fGI/oe7M6ENU6kUNCn7QS8Fv8CfH1R/jfs1/zb4pO1F/wzF2ftN)

The unit tests aren't really testing this, as the effect isn't wrapped around the method call.

I haven't confirmed this, but I suspect these methods could reuse `TrackOpTypes.ITERATE`.